src/Symfony/Component/Console/Descriptor/JsonDescriptor.php

```diff
@@ -115,20 +119,26 @@ private function getInputOptionData(InputOption $option, bool $negated = false):
             'accept_value' => $option->acceptValue(),
             'is_value_required' => $option->isValueRequired(),
             'is_multiple' => $option->isArray(),
+            'is_deprecated' => $option->isDeprecated(),
             'description' => preg_replace('/\s*[\r\n]\s*/', ' ', $option->getDescription()),
             'default' => \INF === $option->getDefault() ? 'INF' : $option->getDefault(),
         ];
+        if (!$option->isDeprecated()) {
+            unset($data['is_deprecated']);
```

Comment:
I would not unset this. The json is easier to use if we don't have conditional keys in it.